yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,107 @@
+from .common import InfoExtractor
+from ..utils import (
+    urlencode_postdata,
+    determine_ext,
+)
+
+
+class FreeTvIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?freetv\.com/(?:peliculas|series)/(?:[^/]+/)*(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://www.freetv.com/peliculas/atrapame-si-puedes/',
+        'md5': 'dc62d5abf0514726640077cd1591aa92',
+        'info_dict': {
+            'id': '428021',
+            'title': 'Atrápame Si Puedes',
+            'full_title': 'Atrápame Si Puedes',
+            'description': 'md5:ca63bc00898aeb2f64ec87c6d3a5b982',
+            'ext': 'mp4',
+        }
+    }, {
+        'url': 'https://www.freetv.com/peliculas/monstruoso/',
+        'md5': '509c15c68de41cb708d1f92d071f20aa',
+        'info_dict': {
+            'id': '377652',
+            'title': 'Monstruoso',
+            'full_title': 'Monstruoso',
+            'description': 'md5:333fc19ee327b457b980e54a911ea4a3',
+            'ext': 'mp4',
+        }
+    }]
+
+    def _extract_video(self, contentId, action="olyott_video_play"):
+        request_body = {
+            'action': action,
+            'contentID': contentId,
+        }
+
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            contentId,
+            'Downloading %s video JSON' % contentId,
+            data=urlencode_postdata(request_body),
+            fatal=False
+        )
+
+        if response is False:
+            return False
+
+        if response.get('success') is False:
+            return False
+
+        response_data = response.get('data')
+
+        if response_data is False:
+            return False
+
+        video_metadata = response_data.get('displayMeta')
+
+        video_id = video_metadata.get('contentID')
+        title = video_metadata.get('title')
+        full_title = video_metadata.get('fullTitle')
+        description = video_metadata.get('desc')
+        video_url = video_metadata.get('streamURLVideo')
```

Comment:
1. Get rid of variables that are only used once
2. Lack of video id/URL should throw error. So make their extraction fatal